### DISCUSSION TITLE
how to test multiple query param in a single request using openapi

### DISCUSSION THREAD

**Discussion by starfrogsplash** [specmatic/specmatic#1069] [Ask For Help] [answered]
```
openapi: 3.0.1
info:
  title: Contract for the Storm Service
  version: '1'
paths:
  /storm/schedule:
    get:
      summary: Get schedule by region, game id, or group id
      security:
        - ApiKeyAuth: []
      parameters:
        - name: region
          in: query
          required: true
          schema:
            type: string
          examples:
            US_200_OK:
              value: 'US'
            INVALID_ID:
              value: 'INVALID'
        - name: weatherId
          in: query
          required: false
          schema:
            type: string
          examples:
            WEATHERINVALID_404:
              value: 'asjn123'
      responses:
        '200':
          description: Get schedule by region, game id, or group id
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/200ResponseBody"
              examples:
                US_200_OK:
                  value:
                    regions: ['US']
                    sport: null
                    league: []
                    items: []
                INVALID_ID: 
                  value:
                    regions: ['INVALID']
                    sport: null
                    league: []
                    items: []
        '404':
          description: Not found
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/ErrorResponseBody"
              examples:
                WEATHERINVALID_404:
                  value: 'weatherIds provided in event not found.'
components:
  securitySchemes:
    ApiKeyAuth:
      type: apiKey
      in: header
      name: X-API-KEY
  schemas:
    ErrorResponseBody:
      type: string
    200ResponseBody:
      type: object
      required:
        - region
      properties:
        regions:
          type: array
          items:
            type: string
        items:
          type: array
          items:
            type: string
```

im trying to test against multiple queries in a single request, but right now specmatic makes 3 seperate requests 2 for 200 requests and 1 for 404, 

for the 404 request i want it to make request using multiple query params that i define, i can set the weatherid but how do i also set the `region` for that 404 request?

```
 npx specmatic test service.yaml --testBaseURL={my_url}
```

**Comment by starfrogsplash** [comment, marked as answer]
figured it out, 

its not mentioned in the docs,

but you have to give the same name for the parameters to be used in the same request

**Reply by yogeshnikam671** [reply]
Hey, it's great that you figured this out. Is there anything in particular you want us to add in the documentation so that it would have helped you resolve this ? 
We would add that info in the documentation so that others can also benefit from the same if they ever come across such issue.

**Reply by starfrogsplash** [reply]
in the documentation there was no mention on how to send multiple params in a single request, which would have helped on how to set it up using the openapi yaml 

also the docs don't really mention that `examples` in `openapi` are what specmatic uses for tests, i.e examples are your tests

**Reply by yogeshnikam671** [reply]
Thanks for responding! We will try to address these points in the documentation. Thanks.

**Reply by yogeshnikam671** [reply]
Hi @starfrogsplash, this is actually mentioned in the documentation. 
You can refer to this link - https://specmatic.in/getting_started.html#how-does-this-all-work

Let me know if this does not clarify about how Specmatic makes use of the examples as tests. We will incorporate your feedback to make the documentation more easier to understand. Thanks.

**Reply by starfrogsplash** [reply]
@yogeshnikam671 

the documentation doesn't explicitly state how Specmatic utilizes examples from the OpenAPI YAML file to generate tests.

`
named example SCOOBY_200_OK listed under the request parameters
`

as you can see it says `SCOOBY_200_OK` under the request paramters,

it doesnt tell you also use the named example are used in the path paramaters to tie to your request paramters

**Reply by yogeshnikam671** [reply]
@starfrogsplash Thanks for pointing this out. 
We are working on updating the documentation.
Will let you know once done.
